### DISCUSSION TITLE
Tags page not able to be opened with database error

### DISCUSSION THREAD

**Discussion by elizhyu** [firefly-iii/firefly-iii#8523] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

All of a sudden, the tags page can not be opened, but instead showing an error page says:
> Whoops! A database error occurred.
The error is:
```
SQLSTATE[22008]: Datetime field overflow: 7 ERROR: date/time field value out of range: "99-01-01 00:00:00" HINT: Perhaps you need a different "datestyle" setting. (Connection: pgsql, SQL: select * from "tags" where "tags"."user_id" = 1 and "tags"."user_id" is not null and "tags"."date" >= 99-01-01 00:00:00 and "tags"."date" <= 99-12-31 23:59:59 and "tags"."deleted_at" is null order by "tags"."tag" asc)
```
This error occured in file `/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php` on line `829` with code `22008`.

### Debug information

Debug information generated at 2024-02-05 14:25:25 for Firefly III version **v6.1.9**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.9 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.2 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td>mysql, <strong>*pgsql*</strong>, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>957, base <span>#</span>79</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>US/Central + America/Chicago</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>info, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-02-04 00:00:00 (1 day ago)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has liabilities">:credit_card:</span> <span title="Has piggy banks">:pig:</span> <span title="Has reconciled">:ledger:</span> <span title="Has object groups">:bookmark_tabs:</span></td></tr><tr><td>Session start</td><td>2024-01-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-12-31 23:59:59</td></tr><tr><td>View range</td><td>1Y</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0</td></tr></tbody></table>

### Steps to reproduce

Click Tags on left side bar at any time any page.

### Additional info

Other Classification pages (Categories, Groups) load with no problem. Only Tags page having the issue.

**Comment by JC5** [comment]
You probably have one tag with an invalid date. You will have to delete it, either manually from the database, or by browsing tags one by one until you find it.
